internal/resource/config/authenticationpolicies/settings/authentication_policies_settings_resource.go

```diff
@@ -0,0 +1,227 @@
+package authenticationpoliciessettings
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/booldefault"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	client "github.com/pingidentity/pingfederate-go-client/v1125/configurationapi"
+	"github.com/pingidentity/terraform-provider-pingfederate/internal/resource/common/id"
+	"github.com/pingidentity/terraform-provider-pingfederate/internal/resource/config"
+	internaltypes "github.com/pingidentity/terraform-provider-pingfederate/internal/types"
+)
+
+// Ensure the implementation satisfies the expected interfaces.
+var (
+	_ resource.Resource                = &authenticationPoliciesSettingsResource{}
+	_ resource.ResourceWithConfigure   = &authenticationPoliciesSettingsResource{}
+	_ resource.ResourceWithImportState = &authenticationPoliciesSettingsResource{}
+)
+
+// AuthenticationPoliciesSettingsResource is a helper function to simplify the provider implementation.
+func AuthenticationPoliciesSettingsResource() resource.Resource {
+	return &authenticationPoliciesSettingsResource{}
+}
+
+// authenticationPoliciesSettingsResource is the resource implementation.
+type authenticationPoliciesSettingsResource struct {
+	providerConfig internaltypes.ProviderConfiguration
+	apiClient      *client.APIClient
+}
+
+// GetSchema defines the schema for the resource.
+func (r *authenticationPoliciesSettingsResource) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	schema := schema.Schema{
+		Description: "Manages Authentication Policies Settings",
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				Description: "The ID of the resource.",
+				Computed:    true,
+				Optional:    false,
+			},
+			"enable_idp_authn_selection": schema.BoolAttribute{
+				Description: "Enable IdP authentication policies.",
+				Optional:    true,
+				Computed:    true,
+				Default:     booldefault.StaticBool(false),
+			},
+			"enable_sp_authn_selection": schema.BoolAttribute{
+				Description: "Enable SP authentication policies.",
+				Optional:    true,
+				Computed:    true,
+				Default:     booldefault.StaticBool(false),
+			},
+		},
+	}
+
+	id.ToSchema(&schema)
+	resp.Schema = schema
+}
+
+func addOptionalAuthenticationPoliciesSettingsFields(ctx context.Context, addRequest *client.AuthenticationPoliciesSettings, plan authenticationPoliciesSettingsModel) error {
+
+	if internaltypes.IsDefined(plan.EnableIdpAuthnSelection) {
```

Comment:
No need for the IsDefined checks here - ValueBoolPointer already will return the correct values if one of these isn't defined (and they should be defined anyways since they both have a default value).